workflows/phaseimpute/main.nf

```diff
@@ -97,31 +99,44 @@ workflow PHASEIMPUTE {
     if (params.steps.split(',').contains("simulate") || params.steps.split(',').contains("all")) {
         // Output channel of simulate process
         ch_sim_output = Channel.empty()
+        // Set truth channel
+        ch_input_validate_truth = ch_input_sim
 
         // Test if the input are all bam files
         getAllFilesExtension(ch_input_sim)
             .map{ if (it != "bam") {
                 error "All input files must be in BAM format to perform simulation"
             } }
 
-        // Split the bam into the region specified
-        BAM_REGION(ch_input_sim, ch_region, ch_fasta)
-        ch_versions = ch_versions.mix(BAM_REGION.out.versions)
+        // Compute coverage of input files
+        SAMTOOLS_COVERAGE_TRT(ch_input_sim, ch_fasta)
+        ch_versions      = ch_versions.mix(SAMTOOLS_COVERAGE_TRT.out.versions)
+        ch_multiqc_files = ch_multiqc_files.mix(SAMTOOLS_COVERAGE_TRT.out.coverage.map{it[1]})
 
-        // Initialize channel to impute
-        ch_bam_to_impute = Channel.empty()
+        if (params.sim_by_reg == true) {
+            // Split the bam into the region specified
+            BAM_REGION(ch_input_sim, ch_region, ch_fasta)
+            ch_versions  = ch_versions.mix(BAM_REGION.out.versions)
+            ch_input_dwn = BAM_REGION.out.bam_region
+        } else {
+            ch_input_dwn = ch_input_sim
+                .map{ meta, bam, index -> [ meta + [chr: "all"], bam, index ] }
```

Comment:
Is there a specific reason for adding a new channel? We could simply overwrite the `ch_input_sim` with the `BAM_REGION.out.bam_region`. Therefore, the `else` would not be needed.

Reply:
I find that doing so add clarity to the script.
And we did it like that for the moment.
We should ask the nf-core community about the best practice on that.